src/constants/routes.ts

```diff
@@ -54,6 +68,17 @@ export const routes = {
   preferences: `${paths.preferences}/:tab?`,
   profilePreferences: [`${paths.preferences}/${PreferencesTabRoutes.Profile}`],
   projectPatches: `${paths.project}/:id/${PageNames.Patches}`,
+  projectSettings: `${paths.project}/:id/${PageNames.Settings}/:tab?`,
+  projectSettingsAccess: `${paths.project}/:id/${PageNames.Settings}/${ProjectSettingsTabRoutes.Access}`,
+  projectSettingsGeneral: `${paths.project}/:id/${PageNames.Settings}/${ProjectSettingsTabRoutes.General}`,
+  projectSettingsGitHubCommitQueue: `${paths.project}/:id/${PageNames.Settings}/${ProjectSettingsTabRoutes.GitHubCommitQueue}`,
+  projectSettingsEventLog: `${paths.project}/:id/${PageNames.Settings}/${ProjectSettingsTabRoutes.EventLog}`,
+  projectSettingsNotifications: `${paths.project}/:id/${PageNames.Settings}/${ProjectSettingsTabRoutes.Notifications}`,
+  projectSettingsPatchAliases: `${paths.project}/:id/${PageNames.Settings}/${ProjectSettingsTabRoutes.PatchAliases}`,
+  projectSettingsPeriodicBuilds: `${paths.project}/:id/${PageNames.Settings}/${ProjectSettingsTabRoutes.PeriodicBuilds}`,
+  projectSettingsProjectTriggers: `${paths.project}/:id/${PageNames.Settings}/${ProjectSettingsTabRoutes.ProjectTriggers}`,
+  projectSettingsVariables: `${paths.project}/:id/${PageNames.Settings}/${ProjectSettingsTabRoutes.Variables}`,
+  projectSettingsVirtualWorkstation: `${paths.project}/:id/${PageNames.Settings}/${ProjectSettingsTabRoutes.VirtualWorkstation}`,
```

Comment:
Woah, So many routes. 
Maybe we could clean this up and extract these into a separate object or separate the route slugs into their own variable so we can reuse it or even a combination of the two.

```
const projectSettingsRoutes = {
  projectSettingsAccess: `${paths.project}/:id/${PageNames.Settings}/${ProjectSettingsTabRoutes.Access}`,
  projectSettingsGeneral: `${paths.project}/:id/${PageNames.Settings}/${ProjectSettingsTabRoutes.General}`,
  projectSettingsGitHubCommitQueue: `${paths.project}/:id/${PageNames.Settings}/${ProjectSettingsTabRoutes.GitHubCommitQueue}`,
  ...
}

export const routes = {
  preferences: `${paths.preferences}/:tab?`,
  profilePreferences: [`${paths.preferences}/${PreferencesTabRoutes.Profile}`],
  projectPatches: `${paths.project}/:id/${PageNames.Patches}`,
  ...projectSettingsRoutes,
}
```
or
```
const projectSettingsSlug = `${paths.project}/:id/${PageNames.Settings}`
...
projectSettingsAccess: `${projectSettingsSlug}/${ProjectSettingsTabRoutes.Access}`,
projectSettingsGeneral: `${projectSettingsSlug}/${ProjectSettingsTabRoutes.General}`
```

What do you think?